### ISSUE TITLE
There is an error with Facebook Ads Extension setup (...)

### ISSUE THREAD

**Issue by antisocialbrother** [facebookincubator/Facebook-For-OpenCart#2]
Version 1.0.13.

There is this error right after the setup:

Facebook Ads Extension setup. Please click on Facebook Ads Extension, Manage Settings, go to Advanced options and click on Delete Settings to restart the setup

log error:

Complete - Get Sample product feed
2018-05-16 16:48:09 - Failure - facebook feed not created, get initial product sync status
2018-05-16 16:48:09 - null

**Comment by shue249** [comment]
Thank you for your feedback. 
1. Will you be able to try to delete away the settings and re-setup Facebook Ads Extension again? To delete away the settings, click on the Manage Settings button, and on the popup window, choose Advanced options and click on Delete Settings.

2. Will you be able to send the log error file to ads_extension_opencart@fb.com?

**Comment by antisocialbrother** [comment]
Yes, that was the first thing I did, same error. I've send the log file.

Thank you.

**Comment by shue249** [comment]
Thank you for your feedback! We are working on a fix now.

**Comment by neonaresh** [comment]
Facing the same issue from over a month, have installed latest versions atleast 10 times, have sent error logs to ads_extension_opencart@fb.com email. Deleted settings and re-configured it a lot of times.

waiting for the resolution

**Comment by shue249** [comment]
Thank you for your feedback and we really appreciate your time and effort with us on improving the plugin. We are releasing out a new version within this week to address this problem.

I do have a question. How many enabled products do your store have now? Our fix is to adopt the API approach when we detect that the feed fails for the initial product sync. However, this method will take more time to do the initial sync.

**Comment by neonaresh** [comment]
I have 4 Lakh enabled products on my store, right now, and we add about new 1500 products each day.

**Comment by shue249** [comment]
In this case, the new fix will still not work, as it will take too long to sync in all the products. 

Earlier, we reached out to you on a test for us to deep dive into the problem. Wondering if you will be open to this test. The process will be as follow:
1. We will send you a alpha version which contains more logging information
2. We ask for your help to do another setup again, most likely the setup will fail.
3. Send to us the log file
4. We will look at the extra logging to see if we can figure out the problem

Let us know your thoughts

**Comment by neonaresh** [comment]
Will it cause any downtime on my live website ? are there any chances ?

If i give you a test server, replicating my present database, will you be able to develop and test your module on that server (with same number of products)

The problem is, communication takes a very long time, i get replies and feedbacks in 24-48 or sometimes 4-5 days. If in case a problem occurs on the website due to testing of this module, and i dont get response back from you, it will cause me huge loss of branding and sales. 

If i replicate my website on another server, so will you be able to test your module on that server ? so that my present website remains safe !

**Comment by shue249** [comment]
No, there will be no downtime on the live website. It will be the exact same plugin version 1.0.13 with few additional logging to show out the errors when the feed upload fail. We will like to keep the exact same server as we suspect there could also be specific server configuration.

**Comment by neonaresh** [comment]
oh okay, Then there is no problem, sure, lets go ahead with this.

**Comment by shue249** [comment]
Here is the alpha version of 1.0.13 with additional logging.
[facebook_ads_extension_113a.ocmod.zip](https://github.com/facebookincubator/Facebook-For-OpenCart/files/2028875/facebook_ads_extension_113a.ocmod.zip)

**Comment by neonaresh** [comment]
Getting this error, while installation 

Modification Facebook Ads Extension is using the same ID code as the one you are trying to upload!

**Comment by antisocialbrother** [comment]
@shue249   ALPHA (113a) version is working fine. I have around 500 products.

@neonaresh Make sure you remove the old version. Then reset settings.

**Comment by shue249** [comment]
@antisocialbrother Can you share the OpenCart log file with me? You can send it to ads_extension_opencart@fb.com. Thanks!

**Comment by antisocialbrother** [comment]
I'm sorry, you mean the facebook_ads_extension.log, correct?

**Comment by shue249** [comment]
Yes that is correct. Apologies for not making myself clear. Its the same log file where you pasted the contents in the very first post above.

**Comment by shue249** [comment]
@antisocialbrother , we double checked your log file. The setup is successful. Things should be working now. Also, when we release any new versions subsequently, feel free to update to the latest version.

@neonaresh , we need some time to check out the log file. At the same time, we suspect it may be due to the large number of products on your database. Following back to your earlier suggestion to create a test server and replicate the large number of products. Will that be too much to ask if we request for this setup? We can test out a fix which loads the products by segments which may be the fix for this. Love to hear your thoughts.

**Comment by neonaresh** [comment]
I have shared the required details, on the email, please kindly check and let me know if any other information required.

**Comment by shue249** [comment]
@zvekov Thank you for your feedback. Can we ask for you to create a separate issue for this so that we can track this. This is on our radar and we are working on a fix for this.

**Comment by shue249** [comment]
@zvekov We have released a new version of the plugin, v2.0.0 which addresses this problem. Will you like to install this new version.

**Comment by neonaresh** [comment]
Hey, the testing server is same as you asked for. Its completely separate than my live server.

Any database, or any file of testing server will not affect my live server.

It has around 3 lakh products, please just give it a try and see if your module works ! 

If there is any other way to list 4 lakh products, for the first time, do let me know !

**Comment by antisocialbrother** [comment]
@shue249 

With version 2.0.0
HTTP ERROR 500 on facebook extension page and control panel.

**Comment by shue249** [comment]
@antisocialbrother Thank you again for your feedback, patience and time with us.

Version 2.0.0 provides support for OpenCart v3 and at the same time, there are path folder location changes for the files in the plugin. The biggest change is that the files are now in the /extension folder instead of the /facebook folder. Can you check if the extension folders are given write+execute access rights. Also, is there anyway you will be able to obtain the webserver log file and send to us at ads_extension_opencart@fb.com?

**Comment by antisocialbrother** [comment]
Yes access rights are ok on these folders.

Also there are no errors in the log files (opencart / httpd server side)
I had the php error display off, when I turn it on the facebook extension page is just empty (OC control panel too - any other menu and the frontend are working fine)

I'll speak with my admin soon for more information but for now I see no errors.

**Comment by shue249** [comment]
When you access the admin/controller/extension folder on your webserver, do you see these files?
facebookadsextension.php
facebookproduct,php
facebookproductfeed.php
facebookproducttrait.php

**Comment by shue249** [comment]
Also, it's weird that there is no info on your webserver log. Usually error 500 will show up something on the server log.

**Comment by antisocialbrother** [comment]
Yes these files are there.

Still no errors.

Back to 113a and it's working again.

**Comment by shue249** [comment]
Apart from the admin/controller, there are also other files which are modified in the plugin. 
![image](https://user-images.githubusercontent.com/11957585/41503526-a5d36ce2-718b-11e8-9905-0f46daaec76d.png)

**Comment by neonaresh** [comment]
Hello @shue249,

How can i proceed with uploading my products to Facebook catalog ? is there any other way to do that ?

**Comment by antisocialbrother** [comment]
@shue249 

All files are there. It's very hard to find out what's the problem without visible errors. 

I'm sending you the opencart log file. I can't see anything useful but you can check it.

**Comment by neonaresh** [comment]
2018-06-17 0:14:00 - Update products for availability change - 258709,259278
2018-06-17 0:14:00 - Failure - facebook feed not created, update products for availability change
2018-06-17 0:14:01 - {"success":true}
2018-06-17 0:42:55 - Failure - facebook feed not created, dashboard module
2018-06-17 0:42:55 - {"success":true}
2018-06-17 0:43:33 - Get Sample product feed
2018-06-17 0:43:47 - Complete - Get Sample product feed
2018-06-17 0:43:51 - Failure - facebook feed not created, get initial product sync status
2018-06-17 0:43:52 - {"success":true}

**Comment by shue249** [comment]
@neonaresh We are still checking on your issue as this is separate issue. Please do bear with us

**Comment by shue249** [comment]
@antisocialbrother The OpenCart log file is the install log. Do you have the httpd (webserver) error file? If you are using apache, it should be inside the /var/log/apache2/error.log

**Comment by shue249** [comment]
@neonaresh We have separated out your problem into a different ticket - https://github.com/facebookincubator/Facebook-For-OpenCart/issues/9. Let's follow up there

**Comment by shue249** [comment]
@antisocialbrother Do you have permission number for your admin and catalog folders? Will you be able to try to change them to 775?

**Comment by antisocialbrother** [comment]
@shue249 Yes I'm using apache. No related errors in log file.

admin and catalog folders are 755.  I can change them to 775. 

Let me take a snapshot of the server and I'll modify them (the shop is live)

**Comment by antisocialbrother** [comment]
@shue249 Switched to 775, no difference.

**Comment by shue249** [comment]
@antisocialbrother What is the permission of the extension folder and its parents

**Comment by antisocialbrother** [comment]
All folders are 755

**Comment by shue249** [comment]
@antisocialbrother You mentioned that the php error is turned off. Wondering if this can be turned on.

**Comment by antisocialbrother** [comment]
Yup I did switch error display on. (both server side and opencart folder)

Now I have just an empty page (no http error 500)

**Comment by shue249** [comment]
Can I also understand if you have vqmod installed?

**Comment by antisocialbrother** [comment]
Yes I have vqmod.

**Comment by shue249** [comment]
Thanks for the feedback! We think the incompatibility is due to vqmod. Let us look into this now.

For the time being, you can revert back to a version 1.0.14 which is the latest version before 2.0.0. https://github.com/facebookincubator/Facebook-For-OpenCart/releases/tag/v1.0.14

**Comment by antisocialbrother** [comment]
You are correct.

With vqmod disabled it's working fine.

**Comment by shue249** [comment]
@antisocialbrother The latest version v2.0.1 fixes this problem.

**Comment by shue249** [comment]
Closing the case for @antisocialbrother 
We are tracking @neonaresh case on https://github.com/facebookincubator/Facebook-For-OpenCart/issues/9

**Comment by ihos1** [comment]
Installed version 2.1.2. But knocks out the error http://prntscr.com/ltzj1f. Looked at logs, seemingly enough time for request processing. Now in the store about 17,000 products. In the future there will be about 20,000 products.
Facebook ads extension 2.1.2 was installed on a site with a quantity of 200 products, everything is super, and does not work with a large amount.
We can use Facebook ads Extension 2.1.2 or for such a quantity of goods need to look for another solution?
Download the log file https://drive.google.com/open?id=1YsJUR_QsNxqf2k6KZrJkt43_kOWPOxAI.
  Sorry for the English I'm from Ukraine.

**Comment by txwbaliever** [comment]
@shue249 I installed this extension successfully, but it has been checking the status of the item when the item is synced. I tried to reinstall it many times without any change. The following is the error log.
2019-08-05 5:51:57 - Sync all products using feed, feed file generated
2019-08-05 5:52:07 - null
2019-08-05 5:52:07 - Failure - facebook feed not created, sync all products using feed
2019-08-05 5:52:15 - Get Sample product feed
2019-08-05 5:52:15 - Complete - Get Sample product feed
